### PR TITLE
BugFix: update function filtering in model listing

### PR DESCRIPTION
**Pull request by shreyasXplain**

### CODE REVIEW

**Review by krishnadurai — COMMENTED** [review]
Let’s add a test that catches this bug.

**Comment by shreyasXplain** [comment]
> Let’s add a test that catches this bug.
@krishnadurai @thiago-aixplain 
We don't need have info like function or language as attributes of the metric class. We might want to add these info to make this validation possible.

File: aixplain/modules/model.py [hunk 1]
```diff
@@ -76,6 +77,12 @@ def __init__(
         self.additional_info = additional_info
         self.url = config.MODELS_RUN_URL
         self.backend_url = config.BACKEND_URL
+        if "function" in additional_info:
```
Comment (thiago-aixplain) [inline aixplain/modules/model.py:80]: @shreyasXplain to convert a string into Enum, you may do it:

```
self.function = Function(additional_info["function"])
```

File: aixplain/modules/model.py [hunk 2]
```diff
@@ -76,6 +77,12 @@ def __init__(
...
+            self.function = None
```
Comment (thiago-aixplain) [inline aixplain/modules/model.py:81]: Add function in the docstring of model asset:

https://github.com/aixplain/aiXplain/blob/M-5038717082-ModelFilterFix/aixplain/modules/model.py#L46

**Review by thiago-aixplain — CHANGES_REQUESTED** [review]

File: aixplain/modules/model.py (hunk 2 above)
Comment (shreyasXplain) [reply aixplain/modules/model.py:81]: But that needs function in all caps i.e. `Function["TRANSLATION"]` will work but `Function["translation"]` will not.

**Review by thiago-aixplain — APPROVED** [review]

**Pull request merged**